File: pset1/mario/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // TODO: Prompt for start size of pyramid
    int h;
    do
    {
         h = get_int("Height of pyramid: ");
    }
     while (h < 1 || h>8);
    // h is a valid input between 1 & 8
    // represents height of pryramid
    
    
    // for loop to print # for each row
    // no of row = no of #
    for(int i=1; i<=h; i++)
    {
        
         //make pyramid right aligned
         for(int k=1; k<=h-i; k++)
         {
           printf(" ");
         }
       
         //print amount of # needed
         for(int j=1; j<=i; j++)
        {
         printf("#");
        }
      printf("\n");
    }
    
    return 0;
}

File: lab6/mario/less/mario.py
from cs50 import get_int


def main():
    height = get_h("Height: ")
    print_pyramid(height)


# get input 
def get_h(prompt):
    while True:
        h = get_int(prompt)
        # check for valid input 
        if h > 0 and h < 9:
            break
        else:
            print("Please enter a number between 1 and 8")
        
            
    return h


# print ## for pryamid 
def print_pyramid(rows):
    
    for row in range(1, rows + 1):
        #use the same method from c to print have pyramid
        print(" " * (rows - row) + "#" * row)


if __name__ == "__main__":
    main()